pkg/util/clusteragent/clusteragent.go

```diff
@@ -283,10 +282,20 @@ func GetClusterAgentEndpoint() (string, error) {
 }
 
 // Version returns ClusterAgentVersion already stored in the DCAClient
-func (c *DCAClient) Version() version.Version {
+// It refreshes the cached version before returning it if withRefresh is true
+func (c *DCAClient) Version(withRefresh bool) version.Version {
 	c.clusterAgentClientLock.RLock()
 	defer c.clusterAgentClientLock.RUnlock()
 
+	if withRefresh {
+		ver, err := c.getVersion()
+		if err != nil {
+			log.Errorf("failed to refresh cluster agent version")
+		} else {
+			c.clusterAgentVersion = ver
```

Comment:
We need to change the calls above to be `Lock()` and `Unlock()` because now we're writing.

Reply:
good catch, will update it